freeradius3: new package

LucileQ opened the pull request:
Hi,

I've bumped freeradius-server to version 3.0.11 because v2.2.x series is marked End Of Life on http://freeradius.org and I needed support from the community to test EAP-TLS authentication.

I've added a config option to install all modules needed to run the RADIUS server with the default configuration files. Since I didn't used SQL nor LDAP for my test, I've disabled the support.

I've only run the RADIUS server in User mode Linux (x86_64). So I'm not sure how would the package build for other targets.

I'll gladly accept any help to improve this commit.

dangowrt commented:
Please create a new package called `freeradius` instead of modifying `freeradius2`.

LucileQ commented:
I've created a new package "freeradius-server".

I've also disabled LDAP support since I did not test it.

champtar commented:
@LucileQ I think it's better to name it "freeradius3"

champtar commented on libs/libtalloc/libtalloc-cache.txt:
```diff
@@ -0,0 +1,39 @@
+Checking simple C program: OK
```
this file (libtalloc-cache.txt) should be removed

LucileQ replied:
libtalloc uses wafsamba to configure and build.
It cannot cross-compiling whithout either --cross-execute (emulator) or --cross-answers (which contains the configuration answers for the target system). https://wiki.samba.org/index.php/Waf#cross-compiling
The file libtalloc-cache.txt contains theses answers. I do not see other options to build libtalloc.

champtar replied:
I though it was an accident ...
I'm not sure this answer file is true for all arch in OpenWrt/LEDE
(at least kernel version will change)

found a package of samba4 (also use waf) and it doesn't use this answer file
https://github.com/tobiaswaldvogel/openwrt-addpack/blob/master/samba4/Makefile
https://forum.openwrt.org/viewtopic.php?id=54490
instead it uses a long list of CONFIGURE_VARS (same thing but we keep everything in the Makefile)

champtar commented on net/freeradius-server/Makefile:
```diff
@@ -0,0 +1,607 @@
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_SOURCE_URL:=\
+	ftp://ftp.freeradius.org/pub/freeradius/ \
+	ftp://ftp.freeradius.org/pub/freeradius/old/
```
can you put http/https mirrors, ftp is not always available

champtar commented on net/freeradius-server/Makefile:
```diff
@@ -0,0 +1,607 @@
+	ftp://ftp.freeradius.org/pub/freeradius/ \
+	ftp://ftp.freeradius.org/pub/freeradius/old/
+PKG_MD5SUM:=e8a748fe824223ed488cc74a1645023e
+PKG_MAINTAINER:=Daniel Golle <daniel@makrotopia.org>
```
@dangowrt are you ok to maintain freeradius3?

LucileQ replied:
I would rather not.
I have bumped freeradius-server to v3.x since version 2.x has been marked as EOL and I didn't want the work to be wasted, however :
I have used freeradius with a limited set of modules (no SQL, no LDAP) in a specific context.
I would not be able to answer accurately any question related to this package and its modules.
I'm unlikely to use it again in the near futur.

champtar replied:
hi @LucileQ, sorry for my brivety,
PKG_MAINTAINER is still set to @dangowrt, that's why i was asking the question.
We only merge package with a maintainer, and if nobody wants to maintain it I can't merge this PR
Just to be sure I'm not misreading, @LucileQ you don't want to maintain this package?

champtar commented on net/freeradius-server/Makefile:
```diff
@@ -0,0 +1,607 @@
+PKG_LICENSE_FILES:=COPYRIGHT LICENSE
+
+PKG_FIXUP:=autoreconf
+PKG_CHECK_FORMAT_SECURITY:=0
```
is this still needed?

champtar commented on net/freeradius-server/Makefile:
```diff
@@ -0,0 +1,607 @@
+  FREERADIUS_SERVER_OPENSSL \
+  FREERADIUS_SERVER_NOSSL
+
+PKG_CHECK_FORMAT_SECURITY:=0
```
dup

champtar commented on net/freeradius-server/Config.in:
```diff
@@ -0,0 +1,14 @@
+config FREERADIUS_SERVER_NOSSL
+	bool "No SSL support"
+
+config FREERADIUS_SERVER_OPENSSL
```
I think this collide with freeradius2 package, please rename it to FREERADIUS3_

champtar commented on net/freeradius-server/Makefile:
```diff
@@ -0,0 +1,607 @@
+define Package/freeradius-server/conffiles
+/etc/freeradius/clients.conf
+/etc/freeradius/radiusd.conf
+/etc/freeradius/sites/default
```
put conf in freeradius3 ?

LucileQ commented:
Hi @champtar,

Thanks for the review.

I've added myself as the MAINTAINER for both new packages.

I've renamed the package freeradius3, changed the mirror to https://github...

I was not able to compile samba4 with the example you provided (https://github.com/tobiaswaldvogel/openwrt-addpack/blob/master/samba4/Makefile which was for samba version 4.0.x, the other examples from the thread https://forum.openwrt.org/viewtopic.php?id=54490 are using an answer file).
So I've included libtalloc "answer file" in the Makefile.

champtar commented on net/freeradius-server/Makefile:
```diff
@@ -0,0 +1,605 @@
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_VERSION).tar.gz
+PKG_SOURCE_URL:=https://github.com/FreeRADIUS/freeradius3/archive
```
your sed was a bit agressive, url is still "https://github.com/FreeRADIUS/freeradius-server/archive"

champtar commented:
Thanks @LucileQ 
can you also rename the directory? (it's still freeradius-server)

LucileQ commented:
Hi @champtar,
I've renamed the directory, and fixed the package's source path.
I've also updated the intallation paths:
/usr/lib/freeradius -> /usr/lib/freeradius3
/usr/share/freeradius -> /usr/share/freeradius3
Let me know if you see other defects.

champtar commented:
Just compile tested it, thanks a lot @LucileQ

Ricardo1980 commented:
Hi @LucileQ 
Can you add sql support? Like sqlite and mysql... thanks.
